From 76894ce62d55027808c955d96d8c6a1747d4a8a1 Mon Sep 17 00:00:00 2001
From: Rahul Sabnis <rahulsabnis@google.com>
Date: Wed, 6 Apr 2022 18:08:18 +0000
Subject: [PATCH] Removes app access to BluetoothAdapter#setScanMode by
 requiring BLUETOOTH_PRIVILEGED permission.

Bug: 203431023
Test: Manual
Change-Id: I50d5ed327a7c90a3f73a9924e5b2b66310dff76c
(cherry picked from commit e49e5dc377dd39c523c287b71c0831159e2cc6ae)
Merged-In: I50d5ed327a7c90a3f73a9924e5b2b66310dff76c
---
 src/com/android/bluetooth/btservice/AdapterService.java | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/com/android/bluetooth/btservice/AdapterService.java b/src/com/android/bluetooth/btservice/AdapterService.java
index 0f8d8691f..e6494ddab 100644
--- a/src/com/android/bluetooth/btservice/AdapterService.java
+++ b/src/com/android/bluetooth/btservice/AdapterService.java
@@ -1433,6 +1433,8 @@ public class AdapterService extends Service {
                 return false;
             }
 
+            enforceBluetoothPrivilegedPermission(service);
+
             service.mAdapterProperties.setDiscoverableTimeout(duration);
             return service.mAdapterProperties.setScanMode(convertScanModeToHal(mode));
         }
-- 
2.36.0.550.gb090851708-goog

